File: Pythonleetcode/Leetcode70.py
import copy
class Solution(object):
    def climbStairs(self, n):
        def jiecheng(n):
            if n==0:
                return 1
            m=copy.deepcopy(n)
            while n!=1:
                n=n-1
                m=m*n
            return m


        def zuhe(x,y):
            return jiecheng(y)/(jiecheng(x)*jiecheng(y-x))

        sum=0
        if n%2==0:
            k=int(n/2)

            for i in range(k+1):
                sum=sum+zuhe(2*i,k+i)
            return sum


        else:
            k=int((n-1)/2)
            for i in range(k+1):
                sum=sum+zuhe(2*i+1,k+i+1)
            return int(sum)


print(Solution().climbStairs(7))

File: CppPractice/CppLeetcode/Leetcode70.cpp
class Solution {
public:
    int climbStairs(int n) 
    {
        vector<int>dp(n+1,0);
        if(n<=1){return 1;}
        dp[0]=1;
        dp[1]=1;
        for(int i=2;i<dp.size();i++)
        {
            dp[i] = dp[i-1]+dp[i-2];
        }
        return dp[dp.size()-1];
        
    }
};